packages/sanity/src/structure/panes/documentList/sheetList/__tests__/DocumentSheetListSelect.test.tsx

```diff
@@ -0,0 +1,107 @@
+import {beforeEach, describe, expect, it, jest} from '@jest/globals'
+import {type SanityDocument} from '@sanity/types'
+import {studioTheme, ThemeProvider} from '@sanity/ui'
+import {type CellContext} from '@tanstack/react-table'
+import {fireEvent, render, screen} from '@testing-library/react'
+
+import {DocumentSheetListSelect} from '../DocumentSheetListSelect'
+
+const mockToggleSelected = jest.fn()
+const mockSetSelectedAnchor = jest.fn()
+const mockSetRowSelection = jest.fn()
+
+const props = {
+  table: {
+    getSelectedRowModel: () => ({rows: [{index: 100}, {index: 122}]}),
+    setRowSelection: mockSetRowSelection,
+
+    options: {
+      meta: {
+        selectedAnchor: null,
+        setSelectedAnchor: mockSetSelectedAnchor,
+      },
+    },
+  },
+  row: {
+    index: 123,
+    getCanSelect: () => true,
+    getIsSelected: () => null,
+    toggleSelected: mockToggleSelected,
+  },
+} as unknown as CellContext<SanityDocument, unknown>
+
+const renderTest = (renderProps?: Partial<CellContext<SanityDocument, unknown>>) =>
+  render(
+    <ThemeProvider theme={studioTheme}>
+      <DocumentSheetListSelect {...{...props, ...(renderProps || {})}} />
+    </ThemeProvider>,
+  )
+
+describe('DocumentSheetListSelect', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  it('selects only current checkbox if shift key is not pressed', async () => {
+    renderTest()
+
+    const checkbox = screen.getByRole('checkbox')
+
+    expect(checkbox).not.toBeChecked()
+
+    fireEvent.click(checkbox)
+    expect(mockToggleSelected).toHaveBeenCalledTimes(1)
+    expect(mockSetSelectedAnchor).toHaveBeenCalledWith(123)
+  })
+
+  it('unselected current checkbox', async () => {
+    const selectProps = {
+      ...props,
+      row: {...props.row, getIsSelected: () => true},
+    }
+    renderTest(selectProps)
+
+    const checkbox = screen.getByRole('checkbox')
+
+    expect(checkbox).toBeChecked()
+
+    fireEvent.click(checkbox)
+    expect(mockToggleSelected).toHaveBeenCalledTimes(1)
+    expect(selectProps.table.options.meta?.setSelectedAnchor).toHaveBeenCalledWith(null)
+  })
+
+  it('disables the checkbox when row is not selectable', () => {
```

Comment:
what's the reason that a row is not selectable? For a11y, it would be best to not have elements disabled.

Reply:
Oh totally! Currently there is no reason why a row would be selected, but tanstack table does support this functionality, so I thought I'd add a theoretical test case to ensure that any future disabling. In the future we mihgt disable, perhaps whilst we are running a batch publish and don't want the user change row selection. Added here so it's already tested